tests/cpp/cell/test_g2s_copy.cu

```diff
@@ -1,4 +1,4 @@
-#include "cell/copy/mod.hpp"
+#include "cell/copy/dyn_copy.hpp"
```

Comment:
I wonder why such modifications are necessary in order for it to compile successfully?

Reply:
The include order in `copy/mod.hpp` is as follows:

https://github.com/TiledTensor/TiledCUDA/blob/b31db2aa1420b595f4ac01a792c714cd81053d1e/include/cell/copy/mod.hpp#L3-L5

`copy/copy.hpp` is before `dyn_copy.hpp`.

`copy.hpp` includes `<cute/algorithm/copy.hpp>`:

https://github.com/TiledTensor/TiledCUDA/blob/b31db2aa1420b595f4ac01a792c714cd81053d1e/include/cell/copy/copy.hpp#L5

and `dyn_copy.hpp` include `<cute/tensor.hpp>`

https://github.com/TiledTensor/TiledCUDA/blob/b31db2aa1420b595f4ac01a792c714cd81053d1e/include/cell/copy/dyn_copy.hpp#L5

In a conclusion, `<cute/algorithm/copy.hpp>` is included before `<cute/tensor.hpp>`. When I upgrade g++ into 10.5.0., it complains that:

```bash
error: namespace "cute::detail" has no member "is_prefetch"
```

A similar issue can be found here: https://github.com/NVIDIA/cutlass/issues/1508

I am not sure the compilation is successful in g++ 9.4.

Reply:
It seems this might be a bug in **CuTe**, where they haven't handled the dependencies between header files very well.

Reply:
> The include order in `copy/mod.hpp` is as follows:
> 
> https://github.com/TiledTensor/TiledCUDA/blob/b31db2aa1420b595f4ac01a792c714cd81053d1e/include/cell/copy/mod.hpp#L3-L5
> 
> `copy/copy.hpp` is before `dyn_copy.hpp`.
> 
> `copy.hpp` includes `<cute/algorithm/copy.hpp>`:
> 
> https://github.com/TiledTensor/TiledCUDA/blob/b31db2aa1420b595f4ac01a792c714cd81053d1e/include/cell/copy/copy.hpp#L5
> 
> and `dyn_copy.hpp` include `<cute/tensor.hpp>`
> 
> https://github.com/TiledTensor/TiledCUDA/blob/b31db2aa1420b595f4ac01a792c714cd81053d1e/include/cell/copy/dyn_copy.hpp#L5
> 
> In a conclusion, `<cute/algorithm/copy.hpp>` is included before `<cute/tensor.hpp>`. When I upgrade g++ into 10.5.0., it complains that:
> 
> ```shell
> error: namespace "cute::detail" has no member "is_prefetch"
> ```
> 
> Similar issues can be found in: [NVIDIA/cutlass#1508](https://github.com/NVIDIA/cutlass/issues/1508) and [NVIDIA/cutlass#1484](https://github.com/NVIDIA/cutlass/issues/1484)
> 
> I am not sure why the compilation is successful in g++ 9.4.

So does that mean when using copy-related functions later, we won't be able to directly include all the copy-related header files, but can only include one specific header file?

Reply:
I suspect things won't be that bad. I suggest leaving this PR to be merged later. I can examine the include order to make it clearer.